### DISCUSSION TITLE
[Porous Flow] Rate of change of fluid component at each node

### DISCUSSION THREAD

**Discussion by steam-simulator** [idaholab/moose#25101] [Q&A Modules: General] [answered]
Hello,
When using the PorousFlow Action (e.g., `PorousFlowFullySaturated`), we can obtain the rate of change of each fluid component using `save_component_rate_in`. I’m using a Kernel consisting of `PorousFlowMassTimeDerivative` and `PorousFlowAdvectiveFlux`. How do I obtain the rate of change of each component? I think my results (obtained from the variables in the postprocessor) only record the change in components, not their rates. Do I divide my results by the timestep? Please, help.

**Comment by cpgr** [comment, marked as answer]
`save_component_rate_in` makes an AuxVariable(s) with the name(s) given, then uses the `save_in` parameter in the time Kernels to save the contribution to these AuxVariables.

I think that the preferred way now is to use the newer tagging system (https://mooseframework.inl.gov/framework_development/tagging.html) to achieve this.
